### PR TITLE
feature tests for Add goobi workflows to registration & basic item registration in general

### PR DESCRIPTION
**Pull request by jmartin-sul**
taking a shot at addressing https://github.com/sul-dlss/argo/pull/938/files#r101326897

addresses #940 (not sure if it's comprehensive enough to close it)

### CODE REVIEW

File: app/controllers/registration_controller.rb [hunk 1]
```diff
@@ -12,17 +12,20 @@ def tracksheet
     render :plain => pdf.render, :content_type => :pdf
   end
 
-  def workflow_list
-    docs = Dor::SearchService.query(%(id:"#{params[:apo_id]}"))['response']['docs']
+  def workflows_for_apo(apo_id)
+    docs = Dor::SearchService.query(%(id:"#{apo_id}"))['response']['docs']
     result = docs.collect { |doc| doc['registration_workflow_id_ssim'] }.compact
-    apo_object = Dor.find(params[:apo_id], :lightweight => true)
+    apo_object = Dor.find(apo_id, :lightweight => true)
```
Comment (cbeer) [inline app/controllers/registration_controller.rb]: I don't think the `lightweight` option does anything any more.

File: app/controllers/registration_controller.rb [hunk 2]
```diff
@@ -12,17 +12,20 @@ def tracksheet
     render :plain => pdf.render, :content_type => :pdf
   end
 
-  def workflow_list
-    docs = Dor::SearchService.query(%(id:"#{params[:apo_id]}"))['response']['docs']
+  def workflows_for_apo(apo_id)
+    docs = Dor::SearchService.query(%(id:"#{apo_id}"))['response']['docs']
```
Comment (cbeer) [inline app/controllers/registration_controller.rb]: I'm not sure I understand why we do this with weird solr indirection, especially when we load the object below anyway?

**Comment by cbeer** [comment]
Two comments, not directly related to your new test, though.

File: app/controllers/registration_controller.rb (hunk 1 above)
Comment (jmartin-sul) [inline app/controllers/registration_controller.rb]: filed https://github.com/sul-dlss/argo/issues/949

File: app/controllers/registration_controller.rb (hunk 2 above)
Comment (jmartin-sul) [inline app/controllers/registration_controller.rb]: me neither.  filed a new issue to investigate:  https://github.com/sul-dlss/argo/issues/950

**Comment by jmartin-sul** [comment]
@cbeer:  yeah, agreed, we should probably clean up the two things you mentioned.  i'm happy to get rid of lightweight usage pretty much immediately, since it only appears in a couple places, and should be easy to test.  i'm also not sure why we do the other thing you mentioned, but it wouldn't surprise me if we were loading things unnecessarily.

but i'd sort of prefer to fix both under new tickets, since neither thing was affected by this PR or the PR it's augmenting, and since these PRs should make the app better without making the specific things you mention worse.

**Comment by jmartin-sul** [comment]
merging based on @peetucket's general approval, and @cbeer's sign-off on my last comment...

**Pull request merged**